java/build.gradle

```diff
@@ -20,21 +18,15 @@ subprojects {
                 url this.devRepo
             }
         }
-
-        if(new File("/usr/share/maven-repo").exists()) {
-            maven {
-                url "/usr/share/maven-repo"
-            }
-        }
-
         mavenCentral()
     }
 
-    // Only enable infer module path with Java 9 and up
-    if(GradleVersion.current() >= GradleVersion.version('6.4')) {
-        java {
-            modularity.inferModulePath = targetJavaRelease.toInteger() > 8
-        }
+    java {
+        modularity.inferModulePath = true
+        // The corresponding -source and -target javac options are suppressed by gradle when
+        // --release is set through options.compileArgs (see above).
+        sourceCompatibility = "1.${targetJavaRelease}"
```

Comment:
We need to set this with the Java plug-in, the old way of setting this is deprecated.

See: https://docs.gradle.org/8.7/userguide/upgrading_version_8.html#java_convention_deprecation